llama_stack/providers/utils/inference/model_registry.py

```diff
@@ -60,8 +60,18 @@ def get_llama_model(self, provider_model_id: str) -> str:
             return None
 
     async def register_model(self, model: Model) -> Model:
-        model.provider_resource_id = self.get_provider_model_id(
-            model.provider_resource_id
-        )
+        provider_resource_id = self.get_provider_model_id(model.provider_resource_id)
+        if provider_resource_id:
+            model.provider_resource_id = provider_resource_id
+        else:
+            if model.metadata.get("llama_model") is None:
```

Comment:
do we validate that llama-model is a legit llama model?

Reply:
added a check.